### PR TITLE
1.28.0 kind image added to nightly test matrix

### PR DESCRIPTION
**Pull request by rhr323**
This PR adds kubernetes 1.28 (via kind) to the nightly test matrix.

### CODE REVIEW

File: .buildkite/e2e/nightly-main-matrix.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.25.3@sha256:f52781bc0d7a19fb6c405c2af83abfeb311f130707a0e219175677e366cc45d1
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.26.3@sha256:61b92f38dff6ccc29969e7aa154d34e38b89443af1a2c14e6cfbd2df6419c66f
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.27.1@sha256:c44686bf1f422942a21434e5b4070fc47f3c190305be2974f91444cd34909f1b
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.27.1@sha256:c44686bf1f422942a21434e5b4070fc47f3c190305be2974f91444cd34909f1b
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.28.0@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31
```
Comment (pebrc) [inline .buildkite/e2e/nightly-main-matrix.yaml:31]: Have you checked if this image which seems to be build for kind 0.20.0 works with kind 0.17.0 (which seems what we are currently using)

Comment (rhr323) [reply .buildkite/e2e/nightly-main-matrix.yaml:31]: Good point, thanks for the reminder. I have now tested kind 0.17 + k8s 1.28.0 images, installing Eck and a single node Elasticsearch cluster worked.

Tangential question: why did we have kindest/node:v1.27.1 listed twice? Is it ok to remove one of them?

Comment (thbkrkr) [reply .buildkite/e2e/nightly-main-matrix.yaml:31]: https://github.com/elastic/cloud-on-k8s/blob/6196b31dd06aa267fad3d0dc770dad57c526afbd/.buildkite/e2e/nightly-main-matrix.yaml#L30-L32

We use the latest kind version to test ipv6. The second occurence defines the ip family to ipv6 (default is ipv4).

Comment (rhr323) [reply .buildkite/e2e/nightly-main-matrix.yaml:31]: Thanks for the clarification! I guess we would still like to keep that, right? If so, should I add 1.28.0 separately (while also keeping 1.27.1 + ipv4 and 1.27.1 + ipv6)?

Comment (rhr323) [reply .buildkite/e2e/nightly-main-matrix.yaml:31]: 👍  (https://github.com/elastic/cloud-on-k8s/pull/7129/commits/8d16149c05e637192176186ee5a0d7d38b527d13)

Should we also update deployer to use the 1.28 images by default (or should we wait until we see tests pass)?

https://github.com/elastic/cloud-on-k8s/blob/main/hack/deployer/config/plans.yml#L165-L186

**Comment by thbkrkr** [comment]
buildkite test this -f p=kind,t=TestSmoke,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.28.0@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31 -m DEPLOYER_KIND_IP_FAMILY=ipv4,DEPLOYER_KIND_IP_FAMILY=ipv6

File: .buildkite/e2e/nightly-main-matrix.yaml (hunk 1 above)
Comment (thbkrkr) [reply .buildkite/e2e/nightly-main-matrix.yaml:31]: We can test almost anything right now by calling `buildkite test this -f ...`, see https://github.com/elastic/cloud-on-k8s/tree/main/.buildkite/e2e/pipeline-gen.

**Comment by thbkrkr** [comment]
buildkite test this -f p=kind,t=TestSmoke,DEPLOYER_CLIENT_VERSION=0.20.0,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.28.0@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31 -m DEPLOYER_KIND_IP_FAMILY=ipv4,DEPLOYER_KIND_IP_FAMILY=ipv6

// Re-run the test with kind 0.20.0 (DEPLOYER_CLIENT_VERSION).

**Comment by rhr323** [comment]
Hm, so [kind release notes](https://github.com/kubernetes-sigs/kind/releases) do say that 0.20 requires docker 20.

There are also 1.28 images for kind 0.19, so I can update the matrix accordingly, and we can try again.

**Comment by rhr323** [comment]
buildkite test this -f p=kind,t=TestSmoke,DEPLOYER_CLIENT_VERSION=0.19.0,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.28.0@sha256:dad5a6238c5e41d7cac405fae3b5eda2ad1de6f1190fa8bfc64ff5bb86173213 -m DEPLOYER_KIND_IP_FAMILY=ipv4,DEPLOYER_KIND_IP_FAMILY=ipv6

**Comment by thbkrkr** [comment]
Great! We can update the defaults clientVersion and kind.nodeImage in `hack/deployer/config/plans.yml` and also update all imagesin the matrix:
```diff
diff --git a/.buildkite/e2e/nightly-main-matrix.yaml b/.buildkite/e2e/nightly-main-matrix.yaml
index 85607df86..82c374799 100644
--- a/.buildkite/e2e/nightly-main-matrix.yaml
+++ b/.buildkite/e2e/nightly-main-matrix.yaml
@@ -21,14 +21,15 @@
   fixed:
     E2E_PROVIDER: kind
   mixed:
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.21.14@sha256:9d9eb5fb26b4fbc0c6d95fa8c790414f9750dd583f5d7cee45d92e8c26670aa1
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.22.15@sha256:7d9708c4b0873f0fe2e171e2b1b7f45ae89482617778c1c875f1053d4cef2e41
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.23.13@sha256:ef453bb7c79f0e3caba88d2067d4196f427794086a7d0df8df4f019d5e336b61
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.24.7@sha256:577c630ce8e509131eab1aea12c022190978dd2f745aac5eb1fe65c0807eb315
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.25.3@sha256:f52781bc0d7a19fb6c405c2af83abfeb311f130707a0e219175677e366cc45d1
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.26.3@sha256:61b92f38dff6ccc29969e7aa154d34e38b89443af1a2c14e6cfbd2df6419c66f
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.27.1@sha256:c44686bf1f422942a21434e5b4070fc47f3c190305be2974f91444cd34909f1b
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.27.1@sha256:c44686bf1f422942a21434e5b4070fc47f3c190305be2974f91444cd34909f1b
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.22.17@sha256:9af784f45a584f6b28bce2af84c494d947a05bd709151466489008f80a9ce9d5
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.23.17@sha256:f77f8cf0b30430ca4128cc7cfafece0c274a118cd0cdb251049664ace0dee4ff
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.24.13@sha256:cea86276e698af043af20143f4bf0509e730ec34ed3b7fa790cc0bea091bc5dd
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.25.9@sha256:c08d6c52820aa42e533b70bce0c2901183326d86dcdcbedecc9343681db45161
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.26.4@sha256:f4c0d87be03d6bea69f5e5dc0adb678bb498a190ee5c38422bf751541cebe92e
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.27.1@sha256:b7d12ed662b873bd8510879c1846e87c7e676a79fefc93e17b2a52989d3ff42b
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.27.1@sha256:b7d12ed662b873bd8510879c1846e87c7e676a79fefc93e17b2a52989d3ff42b
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.28.0@sha256:dad5a6238c5e41d7cac405fae3b5eda2ad1de6f1190fa8bfc64ff5bb86173213
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.28.0@sha256:dad5a6238c5e41d7cac405fae3b5eda2ad1de6f1190fa8bfc64ff5bb86173213
       DEPLOYER_KIND_IP_FAMILY: ipv6

 - label: gke
diff --git a/hack/deployer/config/plans.yml b/hack/deployer/config/plans.yml
index e486475aa..33cc388c8 100644
--- a/hack/deployer/config/plans.yml
+++ b/hack/deployer/config/plans.yml
@@ -165,24 +165,24 @@ plans:
 - id: kind-dev
   operation: create
   clusterName: eck
-  clientVersion: 0.17.0
+  clientVersion: 0.19.0
   provider: kind
-  kubernetesVersion: 1.27.1
+  kubernetesVersion: 1.28.0
   enforceSecurityPolicies: true
   kind:
     nodeCount: 3
-    nodeImage: kindest/node:v1.27.1@sha256:c44686bf1f422942a21434e5b4070fc47f3c190305be2974f91444cd34909f1b
+    nodeImage: kindest/node:v1.28.0@sha256:dad5a6238c5e41d7cac405fae3b5eda2ad1de6f1190fa8bfc64ff5bb86173213
     ipFamily: ipv4
 - id: kind-ci
   operation: create
   clusterName: kind-ci
-  clientVersion: 0.17.0
+  clientVersion: 0.19.0
   provider: kind
-  kubernetesVersion: 1.27.1
+  kubernetesVersion: 1.28.0
   enforceSecurityPolicies: true
   kind:
     nodeCount: 3
-    nodeImage: kindest/node:v1.27.1@sha256:c44686bf1f422942a21434e5b4070fc47f3c190305be2974f91444cd34909f1b
+    nodeImage: kindest/node:v1.28.0@sha256:dad5a6238c5e41d7cac405fae3b5eda2ad1de6f1190fa8bfc64ff5bb86173213
```

**Comment by rhr323** [comment]
Thanks @thbkrkr, I updated the images as recommended. What else should we test for this change?

**Comment by thbkrkr** [comment]
buildkite test this -f p=kind,DEPLOYER_CLIENT_VERSION=0.19.0,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.28.0@sha256:dad5a6238c5e41d7cac405fae3b5eda2ad1de6f1190fa8bfc64ff5bb86173213 -m DEPLOYER_KIND_IP_FAMILY=ipv4,DEPLOYER_KIND_IP_FAMILY=ipv6

Run the full test suite.

**Comment by thbkrkr** [comment]
buildkite test this -f p=kind,DEPLOYER_CLIENT_VERSION=0.19.0,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.28.0@sha256:dad5a6238c5e41d7cac405fae3b5eda2ad1de6f1190fa8bfc64ff5bb86173213,t=TestAgentConfigRef -m DEPLOYER_KIND_IP_FAMILY=ipv4,DEPLOYER_KIND_IP_FAMILY=ipv6

Run only TestAgentConfigRef which previously failed.

**Review by thbkrkr — APPROVED** [review]

**Review by barkbay — APPROVED** [review]
LGTM

**Comment by pebrc** [comment]
I think this should be backported to 2.9 to verify it works with the already released version as well.

**Pull request merged**